src/Elastic.Markdown/Myst/InlineParsers/DiagnosticLinkInlineParser.cs

```diff
@@ -60,7 +60,7 @@ public override bool Match(InlineProcessor processor, ref StringSlice slice)
 
 		if (Uri.TryCreate(url, UriKind.Absolute, out var uri) && uri.Scheme.StartsWith("http"))
 		{
-			var baseDomain = string.Join('.', uri.Host.Split('.')[^2..]);
+			var baseDomain = uri.Host == "localhost" ? "localhost" : string.Join('.', uri.Host.Split('.')[^2..]);
```

Comment:
I need your help here @Mpdreamz.

This fixes the error. However, I would have expected that it emits a warning because "localhost" is not defined in the `external_hosts` configuration. But it does not.

Reply:
I found this https://github.com/elastic/docs-builder/blob/main/src/Elastic.Markdown/IO/ConfigurationFile.cs#L26 

and I would have expected that I need to add `localhost` here. But it's not needed.

Reply:
Interesting I am getting: 

```
Warning: External URI 'http://localhost:8080/' is not allowed. Add 'localhost' to the 'external_hosts' list in /Users/mpdreamz/Projects/docs-builder/docs/source/docset.yml to allow links to this domain.
    ┌─[/Users/mpdreamz/Projects/docs-builder/docs/source/testing/req.md]
```

You might hit a chicken and egg where you ran docs-builder before and after the fix and the file is not marked as invalid and thus incremental compile is not re-evaluating it.


```
:: info: DocumentationGenerator[0] Re-evaluating /Users/mpdreamz/Projects/docs-builder/docs/source/testing/req.md
```

If you clear the output directory `rm -rf .artifacts/docs/html` or run the tool with the `--force` flag do you see the error message too?

Reply:
Ah!

I only get the warning when I run `dotnet run`

If I run `dotnet run -- serve` it doesn't show the warning.

This could be worth a follow issue, WDYT?

Reply:
Alright, I added `localhost` to the default list https://github.com/elastic/docs-builder/pull/157/commits/021ab63bfe64c5fa9ca5a717e2b5212ac70c469a

Reply:
So with the `serve` command I want to surface the errors in the browser itself rather than having folks needing to dig through the CLI :) 

We'll need a issue for that though

Reply:
I think we can draw a lot of inspiration from existing frameworks with live reload functionality (e.g. nextjs)

And I think they actually do both. Showing the error in the console as well as in the browser.

But we can definitely discuss it in the issue.

Reply:
Follow-up issue: https://github.com/elastic/docs-builder/issues/158